qa/tasks/cephfs/test_volumes.py

```diff
@@ -4345,6 +4337,298 @@ def test_subvolume_user_metadata_list_and_remove_for_legacy_subvolume(self):
+                        'sudo ls volumes/_deleting')
+                except CommandFailedError as e:
+                    if e.exitstatus == 2:
+                        # this means "volume/deleting" directory is absent,
+                        # meaning trash is indeed empty
+                        return
+                    else:
+                        raise
+
+                # ensure all whitespace is removed, else "if" following it
+                # will fail to check if string has no filenames in it.
+                output = output.strip()
+
+                if not output:
+                    break
+
+    def test_when_changed_to_false_unpurged_subvol_is_purged(self):
+        '''
+        Test that when MGR config option mgr/volumes/pause_purging_trash is
+        is set to false, trashed but unpurged subvolume is purged (fully).
+        '''
+        v = self.volname
+        sv = 'sv1'
+
+        self.run_ceph_cmd(f'fs subvolume create {v} {sv} --mode=777')
+        self.config_set('mgr', self.conf_opt, True)
+        sv_path = self._get_sv_path(v, sv)
+        self._do_subvolume_io(sv, number_of_files=10)
+        sv_files = self.mount_a.get_shell_stdout(f'sudo ls {sv_path}').\
+            strip().split('\n')
+
+        self.run_ceph_cmd(f'fs subvolume rm {v} {sv}')
+        self._assert_trashed_sv_is_unpurged(sv, sv_path, sv_files)
+        time.sleep(7)
+        self._assert_trashed_sv_is_unpurged(sv, sv_path, sv_files)
+
+        # XXX actual test here: test that unpurged subvol is purged
+        self.config_set('mgr', self.conf_opt, False)
+        self.wait_for_trashcan_to_be_empty()
+
+    def _get_trashed_sv_path(self, sv, sv_path):
+        uuid = self.mount_a.get_shell_stdout('sudo ls volumes/_deleting').\
+            strip()
+
+        trashed_sv_path = sv_path.replace('_nogroup', f'_deleting/{uuid}')
+        trashed_sv_path = trashed_sv_path.replace(sv, '')
+        return trashed_sv_path
+
+    def _get_num_of_files_in_trashed_sv(self, trashed_sv_path):
+        trashed_sv_files = self.mount_a.get_shell_stdout(
+            f'sudo ls {trashed_sv_path}').strip().split('\n')
+        return len(trashed_sv_files)
+
+    def _assert_trashed_sv_has_num_of_files(self, trashed_sv_path,
+                                            num_of_files):
+        sv_files = self.mount_a.get_shell_stdout(
+            f'sudo ls {trashed_sv_path}').strip().split('\n')
+
+        self.assertEqual(len(sv_files),  num_of_files)
+
+    def test_when_changed_to_true_ongoing_purge_halts(self):
+        '''
+        Test that when MGR config option mgr/volumes/pause_purging_trash is
+        set to true, running "ceph fs subvolume rm" will move the subvolume
+        to trash but (asynchronous) purge of the subvolume won't happen till
+        this config option is not explicitly set to False.
+        '''
+        v = self.volname
+        sv = 'sv1'
+        self.config_set('mgr', self.conf_opt, False)
```

Comment:
Why are we explicitly setting it to `False` when it is already the default value

Reply:
ping?

Reply:
Have you checked the latest code? I don't see this line of code at line number 4566 anymore.